packages/protocol/contracts/L1/libs/LibUtils.sol

```diff
@@ -28,6 +28,23 @@ library LibUtils {
         found = (blk.blockId == id && blk.verifiedForkChoiceId != 0);
     }
 
+    function getForkChoiceId(
+        TaikoData.State storage state,
+        TaikoData.Block storage blk,
+        uint256 blockId,
+        bytes32 parentHash,
+        uint32 parentGasUsed
+    ) internal view returns (uint256) {
+        uint256 fcId = state.forkChoiceIds[blockId][parentHash][parentGasUsed];
+        if (fcId >= blk.nextForkChoiceId) return 0;
```

Comment:
```suggestion
        bytes32 parentHash,
        uint32 parentGasUsed
    ) internal view returns (uint256) {
        uint256 fcId = state.forkChoiceIds[blk.blockId][parentHash][parentGasUsed];
```
Maybe useful to do it like this because this automatically rules out some cases that should not happen and makes sure the input data is consistent.